src/main/groovy/edu/oregonstate/mist/cardsapi/resources/CardsResource.groovy

```diff
@@ -65,4 +81,162 @@ class CardsResource extends Resource {
         }
 
     }
+
+    //Get cards by parameters
+    @GET
+    @Path ('')
+    @Produces(MediaType.APPLICATION_JSON)
+    Response getCards(@QueryParam("types") List<String> types,
+                      @QueryParam("name") Optional<String> name,
+                      @QueryParam("colors") List<String> colors,
+                      @QueryParam("rarities") List<String> rarities,
+                      @QueryParam("energyMin") Optional<Integer> energyMin,
+                      @QueryParam("energyMax") Optional<Integer> energyMax,
+                      @QueryParam("keywords") List<String> keywords,
+                      @QueryParam("number") Optional<Integer> number,
+                      @QueryParam("isRandom") Optional<Boolean> isRandom) {
+
+        Response response
+
+        List<String> validTypes = ["skill", "attack", "power", "status", "curse"]
+        List<String> validColors = ["red", "green", "blue", "colorless"]
+        List<String> validRarities = ["basic", "common", "uncommon", "rare"]
```

Comment:
Maybe these should be static class level variables? I can see `POST` and `PUT` using these lists.